components/tidb_query/src/expr/builtin_string.rs

```diff
@@ -960,6 +960,28 @@ impl ScalarFunc {
         result.push(b'\'');
         Ok(Some(Cow::Owned(result)))
     }
+
+    // see https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_ord
+    #[inline]
+    pub fn ord<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<i64>> {
+        let s = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
+        let bytes = if let Some((n, _)) = s.char_indices().nth(1) {
+            &s.as_bytes()[..n]
+        } else {
+            s.as_bytes()
+        };
+        let mut result = 0;
+        let mut factor = 1;
+        for b in bytes.iter().rev() {
```

Comment:
According to MySQL documentation 
```
(1st` byte code)
+ (2nd byte code * 256)
+ (3rd byte code * 256^2) ...
```
I think you should not reverse the slice.

Reply:
@lonng Thanks for your review.
I hold the same opinion at first. But [the go version](https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L2715)  reverses the bytes and the reversed result seems to be true.

Reply:
I cannot image that the result can be compatible with MySQL, can you check the result of MySQL? And we should fix the behavior of TiDB if it’s not compatible with MySQL.

Reply:
Tested with test cases in https://github.com/pingcap/tidb/blob/master/expression/builtin_string_test.go#L2004. It's the right behavior.
```text
mysql> select ord('你好');
+---------------+
| ord('你好')   |
+---------------+
|      14990752 |
+---------------+
1 row in set (0.00 sec)

mysql> select ord('にほん');
+------------------+
| ord('にほん')    |
+------------------+
|         14909867 |
+------------------+
1 row in set (0.00 sec)
```

The algorithm seems like just reinterpret the bytes into an integer ( like `i64::from_be_bytes(bytes)` ) ( maybe I can do some optimize? )
The "order" in MySQL document may mean order in little-endian.